lib/puppet/provider/sensu_plugin/sensu_install.rb

```diff
@@ -0,0 +1,152 @@
+Puppet::Type.type(:sensu_plugin).provide(:sensu_install) do
+  desc "Provider sensu_check using sensuctl"
+
+  mk_resource_methods
+
+  commands :gem => '/opt/sensu-plugins-ruby/embedded/bin/gem'
+  commands :sensu_install => 'sensu-install'
+
+  def self.instances
+    plugins = []
+
+    output = gem('list', '--local', '^sensu-(plugins|extensions)')
+    Puppet.debug("gem output: #{output}")
+    output.each_line do |o|
+      next unless o.start_with?('sensu-')
+      if o =~ /^(\S+)\s+\((.+)\)/
```

Comment:
please document this regex in plain english

Reply:
Added comments.